src/main/kotlin/com/nfu/gateway/filter/jwt/JwtFilter.kt

```diff
@@ -0,0 +1,81 @@
+package com.nfu.gateway.filter.jwt
+
+import com.nfu.gateway.constant.header.HeaderName
+import com.nfu.gateway.constant.jwt.JwtPrefix
+import com.nfu.gateway.exception.ApiException
+import com.nfu.gateway.exception.constant.ApiError
+import com.nfu.gateway.util.jwt.JwtPayloadParser
+import com.nfu.gateway.util.jwt.JwtValidator
+import com.nfu.gateway.util.startsWithNot
+import io.github.oshai.kotlinlogging.KotlinLogging
+import org.springframework.cloud.gateway.filter.GatewayFilter
+import org.springframework.cloud.gateway.filter.factory.AbstractGatewayFilterFactory
+import org.springframework.http.HttpHeaders
+import org.springframework.http.server.reactive.ServerHttpRequest
+import org.springframework.stereotype.Component
+
+/**
+ * Gateway JWT Filter
+ *
+ * @author baetaehyeon
+ * @since v1.0.0
+ */
+@Component
+class JwtFilter(
+    private val jwtValidator: JwtValidator,
+    private val jwtPayloadParser: JwtPayloadParser,
+): AbstractGatewayFilterFactory<JwtFilter.Config>(Config::class.java) {
+
+    private val logger = KotlinLogging.logger {}
+
+    class Config
+
+    override fun apply(config: Config): GatewayFilter {
+        return GatewayFilter { exchange, chain ->
+            logger.debug { "[JwtFilter] Start" }
+
+            val authorizationHeader = extractAuthorizationHeader(exchange.request)
+            val token = extractToken(authorizationHeader)
+            verifyToken(token)
+
+            val payload = jwtPayloadParser.parse(token)
+            addNfuMemberIdHeader(exchange.request, memberId = payload.id)
+
+            logger.debug { "[JwtFilter] token: $token" }
+            logger.debug { "[JwtFilter] memberId: ${payload.id}" }
+            logger.debug { "[JwtFilter] memberEmail: ${payload.email}" }
+            logger.debug { "[JwtFilter] End" }
+            return@GatewayFilter chain.filter(exchange)
+        }
+    }
+
+    private fun verifyToken(token: String) {
+        if (jwtValidator.isInvalid(token)) {
+            throw ApiException(ApiError.UNAUTHORIZED)
+        }
+    }
+
+    private fun extractToken(authorizationHeader: String): String {
+        if (authorizationHeader.isNullOrBlank() || authorizationHeader.startsWithNot(JwtPrefix.BEARER)) {
```

Comment:
```suggestion
        if (authorizationHeader.isBlank() || authorizationHeader.startsWithNot(JwtPrefix.BEARER)) {
```
@qoxogus
`authorizationHeader` 타입이 non-nullable 이네요.
제안합니다!

Reply:
원래 있었던 로직이 메서드로 분리되고, 덧붙여지면서 제가 놓쳤네요 !!
확인 감사합니다 ~~ !

[91204e2](https://github.com/T-White-Eyes/nfu-gateway-server/pull/1/commits/91204e2e6f1e745855dd2bd0fee1184498e03127)